### PR TITLE
Add confirmation dialog to Delete in beatmap context menu

### LINKED ISSUE #2059: Deleting from right click context menu on beatmap panel (song select) doesn't confirm
Hitting shift-delete brings up a confirmation dialog; right click should also do this.

### PR DESCRIPTION
**Pull request by swoolcock**
Closes #2059

### CODE REVIEW

**Comment by peppy** [comment]
Trying to decide whether we want to make confirmation dialogs locally or add this to BeatmapMaanger (maybe as a `bool confirm = default` parameter). Thoughts?

**Comment by swoolcock** [comment]
Should `BeatmapManager` care about the UI at all though?  Separation of concerns etc.

**Comment by peppy** [comment]
It already does to an extent, by posting notifications.

**Comment by swoolcock** [comment]
I wonder if it could be useful to move dialog boxes to an enum.  It would keep them all in one spot which would be easier for localisation.

**Comment by smoogipoo** [comment]
I would argue that `BeatmapManager` exposing a `PostNotification` event isn't quite the same as it caring about UI stuff.

I agree with @swoolcock in that `BeatmapManager` shouldn't be doing stuff directly to the UI, and most definitely should not require UI interaction to work. I'm fine with this being re-implemented wherever needed.

File: osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs [hunk 1]
```diff
@@ -89,6 +89,12 @@ private void load(LocalisationEngine localisation, BeatmapManager manager, Beatm
             };
         }
 
+        private void delete(BeatmapSetInfo beatmap)
```
[the remark this answers is not in the document]
Comment (swoolcock) [reply osu.Game/Screens/Select/Carousel/DrawableCarouselBeatmapSet.cs]: I copy pasted from `SongSelect`.  I'll inline it instead.

**Review by peppy — APPROVED** [review]

**Pull request merged**